### ISSUE TITLE
Guard debug-messages with a flag

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#1136]
**Reported by sjoelund&#x2e;se on 28 May 2013 09:52 UTC**
In ModelicaStandardTables.c we currently have:
```c
/* Print info message, that file is loading */
ModelicaFormatMessage("... \"%s\" loading (tables for interpolation)\n", fileName);
```

I consider this a bug because you usually don't want your output filled with expected messages (you basically only want warnings or errors). It's fine for a debug version of the library or with a flag set (`readTable(verbose=true)` or similar).

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/1136

**Comment by modelica-trac-importer** [comment]
**Modified by beutlich on 28 May 2013 09:55 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 28 May 2013 10:00 UTC**
Same message was also thrown by old table implementation in Dymola. That's why it was added here again.

**Comment by modelica-trac-importer** [comment]
**Comment by sjoelund&#x2e;se on 28 May 2013 10:05 UTC**
Copy good behaviour, not bad :) Anyway, OpenModelica did not have this behaviour in the past.

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 28 May 2013 10:09 UTC**
Neither there is such a message in built-in tables of SimulationX.

Anyway, it is beta and lib officer has to decide here.

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 28 May 2013 10:13 UTC**
When defining a table, it is very easy to make a mistake, e.g., a relative file name is given, but the executable is in a non-expected directory and suddenly the wrong table is used, or there is a typo in the table name and the wrong table is used. All this might lead to a simulation where the result is not as expected and the question is how the user should find such a mistake.

My decision in 1999 was (when I made the first table implementation) therefore, to always print when a file is read in order to have more information when something goes wrong.

This is a matter of taste and end-users usually prefer to have such type of information.

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 28 May 2013 10:17 UTC**
Replying to [ticket:1136 sjoelund.se]:
> In ModelicaStandardTables.c we currently have:
> ```c
> /\* Print info message, that file is loading \*/
> ModelicaFormatMessage("... \"%s\" loading (tables for interpolation)\n", fileName);
> ```
> 
> I consider this a bug because you usually don't want your output filled with expected messages (you basically only want warnings or errors). It's fine for a debug version of the library or with a flag set (`readTable(verbose=true)` or similar).

You are right this is a good idea to include a flag "verbose = true" in the argument list of readTable in order that from the Modelica side the output can be controlled. This flag could be set from a parameter in the tables menu (with a default of "true").

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 2 Jun 2013 19:52 UTC**
Resolved by cf59de0f2ba234b4483872d3b0ffb83e4426c7ff.
